tests/test_ssl.py

```diff
@@ -2755,43 +2758,143 @@ def test_set_session_wrong_args(self):
         with pytest.raises(TypeError):
             connection.set_session(object())
 
-    def test_client_set_session(self):
+    def test_session_reused(self):
+        """
+        `Connection.session_reused`, returns 0 for new connections..
+        """
+        ctx = Context(TLSv1_2_METHOD)
+        connection = Connection(ctx, None)
+
+        assert connection.session_reused() == 0
+
+    def test_client_set_session_tls1_2(self):
```

Comment:
I went with both TLS 1.2 and 1.3 tests since the session handling is a big different.

I will refactor the tests to share more code.